velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -269,29 +267,145 @@ class ArrayIntersectExceptFunction : public exec::VectorFunction {
 
   // If there's a `constantSet`, whether it refers to left or right-hand side.
   const bool isLeftConstant_{false};
-};
+}; // class ArrayIntersectExcept
+
+template <typename T>
+class ArraysOverlapFunction : public exec::VectorFunction {
+ public:
+  ArraysOverlapFunction() = default;
+
+  ArraysOverlapFunction(SetWithNull<T> constantSet, bool isLeftConstant)
+      : constantSet_(std::move(constantSet)), isLeftConstant_(isLeftConstant) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    // If one of them is a ConstantVector, iterate through the other
+    // ArrayVector->elementsFlatVector and do constantSet_.find().
+    BaseVector* left = args[0].get();
+    BaseVector* right = args[1].get();
+    if (constantSet_.has_value() && isLeftConstant_) {
+      std::swap(left, right);
+    }
+    exec::LocalDecodedVector arrayDecoder(context, *left, rows);
+    exec::LocalDecodedVector elementsDecoder(context);
+    auto decodedLeftElements =
+        decodeArrayElements(arrayDecoder, elementsDecoder, rows);
+
+    BaseVector::ensureWritable(rows, BOOLEAN(), context->pool(), result);
+    auto resultBoolVector = (*result)->template asFlatVector<bool>();
+    auto processRow = [&](auto row, const SetWithNull<T>& rightSet) {
+      auto decodedLeftArray = arrayDecoder.get();
```

Comment:
Move these 2 lines outside of the processRow as they are common.

Reply:
+1